hmt/serve/mock/faker/stateless_faker.py

```diff
@@ -117,10 +117,10 @@ def process(self, spec: OpenAPISpecification, request: Request) -> Any:
                 return self._empty_response(status_code, new_headers)
 
             faker_data = FakerData(
-                spec=spec,
-                path_item=path_item,
+                spec=cast(OpenAPISpecification, spec),
```

Comment:
Is this cast safe? Are we sure it is not none by this point? Usually, tools like mypy/pyright can pick up on if/then clauses that rule out none, so maybe it's better to use the typing engine that way instead of having a cast?